dist/runtime.js

```diff
@@ -17,6 +17,44 @@ var tornado = {
   helpers: {},
 
   /**
+   * Helper context
+   */
+  helperContext: {
+    push: function push() {
+      this.__contextStack.push({});
+    },
+    pop: function pop() {
+      return this.__contextStack.pop();
+    },
+    peek: function peek() {
+      return this.__contextStack[this.__contextStack.length - 1];
+    },
+    get: function get(key) {
+      for (var i = this.__contextStack.length - 1; i >= 0; i--) {
+        var context = this.__contextStack[i];
+        if (context[key] !== undefined) {
+          return context[key];
+        }
+      }
+    },
+    set: function set(key, val) {
+      var context = this.peek();
+      context[this.normalizeKey(key)] = val;
+    },
+    setGlobal: function setGlobal(key, val) {
+      this.__contextStack[0][this.normalizeKey(key)] = val;
+    },
+    clear: function clear(key) {
+      this.set(key, null);
+    },
+    normalizeKey: function normalizeKey(key) {
+      // Prepend with $ if $ isn't already present
+      return key[0] === "$" ? key : "$" + key;
+    },
+    __contextStack: [{}]
```

Comment:
is this to initialize the global? slightly more explicit if we used `push` instead.
```
// push the global context
push();
```